homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -18,9 +19,9 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 HAP_SUFFIX = "._hap._tcp.local."
-POWERVIEW_SUFFIX = "._powerview._tcp.local."
+POWERVIEW_G2_SUFFIX = "._powerview._tcp.local."
+POWERVIEW_G3_SUFFIX = ".powerview-g3.local."
```

Comment:
This is incorrect for Gen3. Needs to be `._powerview-g3.local.` zeroconf to find it.

Reply:
That contradicts the information from Hunter Douglas provided here

https://community.home-assistant.io/t/hunter-douglas-powerview-gen-3-integration/424836/110?u=kingy444

The discovery you mention will find all hubs and not just the primary as I understand.

Can you provide any input with the info form the above linked in mind?

We wouldn't want to discover the secondary because it doesn't do anything and cannot accept commands (it is basically a repeater)

Reply:
What does it look like in https://apps.apple.com/us/app/discovery-dns-sd-browser/id1381004916?mt=12 ?

Reply:
ios version https://apps.apple.com/us/app/discovery-dns-sd-browser/id305441017

Reply:
If it follows the mdns spec it should be `"._powerview-g3._tcp.local."`

Reply:
I forgot I was only testing with one gateway 🤦 . It does follow the mdns spec with the name being `._powerview-g3._tcp.local.` because I wasn't seeing my gateway until I made that change. But as @kingy444 pointed out that'll show all gateways and we need to ignore secondary gateways. I have a second gateway I can test with today on this home. Its unclear to me if there is a unique name for the primary that we can filter on, or if the flow has to be detect that n number of gateways exist on the network, and then use `powerview-g3.local` to communicate to the primary.

Reply:
Discovery output:
<img width="221" alt="Screenshot 2023-08-21 at 08 47 15" src="https://github.com/home-assistant/core/assets/2565164/65e337a8-63a9-430d-add9-04e75af4c56e">

Reply:
I'm not sure `powerview-g3.local` will always be the primary

https://datatracker.ietf.org/doc/html/rfc6762#page-31

Reply:
Is it possible to have two systems with two primaries on the same network?

Reply:
Possible, yes. But unlikely outside of some really weird testing environments. Two primary gateways means two homes in the PowerView App. You'd end up with this:
![image](https://github.com/home-assistant/core/assets/2565164/215ec958-08d8-4059-82d5-eec1b9f204e8)

It would be more likely to have a Gen2 Hub and Gen3 Gateway in the same home, but they're different mdns suffixes so you wouldn't run into the conflicting names.